### PR TITLE
Port to Python 3

### PR DESCRIPTION
**Pull request by pro-panda**
**Tested on:** Ubuntu 16.04, Sugar 0.112

Using Python 3 version of [Sugar Toolkit](https://github.com/Pro-Panda/sugar-toolkit-gtk3/tree/python3-port)

### CODE REVIEW

**Comment by quozl** [comment]
Reviewed to c2ddc7d

**Comment by quozl** [comment]
Rebased on 9872a32.

File: activity.py [hunk 1]
```diff
@@ -380,12 +380,12 @@ def __open_on_journal(self, widget, url):
...
             'mime_type': 'text/uri-list',
         }
-        for k, v in metadata.items():
+        for k, v in list(metadata.items()):
             jobject.metadata[k] = v
         file_path = os.path.join(get_activity_root(), 'instance',
                                  '%i_' % time.time())
         open(file_path, 'w').write(url + '\r\n')
-        os.chmod(file_path, 0755)
+        os.chmod(file_path, 0o755)
```
Comment (Aniket21mathur) [inline activity.py]: @pro-panda @quozl  python 3 will interpret "0o755" as octal and ultimately 755 as 493. This is a discrepancy that 2to3 creates. We should change "0o755" manually to "0755" or "755".

Comment (quozl) [reply activity.py]: I don't understand what you mean.  Octal is how we usually understand and explain file modes, as they are a bit mask of permissions.  In python3 "0755" correctly gives us _SyntaxError: invalid token_.  Decimal 755 would give an insane file mode.  I'm really glad we don't have to use seven OR'd constants!

Comment (Aniket21mathur) [reply activity.py]: Thanks for clarifying.

**Comment by pro-panda** [comment]
@quozl @aniket21mathur, please review

**Comment by Aniket21mathur** [comment]
Reviewed and tested!

**Comment by quozl** [comment]
Added other commits from _master_ branch.  About to rebase.

**Comment by quozl** [comment]
Tested.

**Pull request merged**